### PR TITLE
Migration of advanced MQTT settings to config entry - restructure of common MQTT docs

### PR DESCRIPTION
**Pull request by jbouwh**
## Proposed change

The advanced MQTT configuration options will migrate to the config entry. The current documentation is scattered over multiple pages making it hard to find. This PR moves the common MQTT documentation parts to the MQTT integration page and cleans up the old pages.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/79351
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by jbouwh** [comment]
https://github.com/home-assistant/home-assistant.io/pull/24636 needs to be merged first, then after a rebase this PR can be merged.

**Review by frenck — CHANGES_REQUESTED** [review]
Nice @jbouwh 👍 
Can we add redirects for the old URLs to the new page? (source/_redirects file)

File: source/_integrations/mqtt.markdown [hunk 1]
```diff
@@ -27,27 +27,753 @@ ha_platforms:
...
+    'init':                'initial',
+    'hum_cmd_t':           'target_humidity_command_topic',
+    'hum_cmd_tpl':         'target_humidity_command_template',
+    'hum_stat_t':          'target_humidity_state_topic',
+    'hum_stat_tpl':        'target_humidity_state_template',
+    'json_attr':           'json_attributes',
+    'json_attr_t':         'json_attributes_topic',
+    'json_attr_tpl':       'json_attributes_template',
+    'l_ver_t':             'latest_version_topic',
+    'l_ver_tpl':           'latest_version_template',    'max_mirs':            'max_mireds',
```
Comment (frenck) [inline source/_integrations/mqtt.markdown]:
```suggestion
    'l_ver_tpl':           'latest_version_template',
    'max_mirs':            'max_mireds',
```

**Review by frenck — APPROVED** [review]
Thanks, @jbouwh 👍

**Pull request merged**